Caleb hill rest api docs

Caleb-Hill opened the pull request:
This PR adds a redoc docker container, moves the rest_api docs from the `core` directory to the `1.2` directory, and fixes the markdown issues found in those docs.

Caleb-Hill committed:
Add redoc container

This commit adds the `sawtooth-docs-redoc` container to the docker
compose file.

Signed-off-by: Caleb Hill <hill@bitwise.io>

Caleb-Hill committed:
Add sawtooth-docs-redoc docker file

This commit adds a docker file to do redoc documentation of the
openapi.yaml file from sawtooth-core.

Signed-off-by: Caleb Hill <hill@bitwise.io>

Caleb-Hill committed:
Add Proxy settings for apache

The apache container needs to be told to proxy connections to the redoc
container at the `docs/1.2/rest_api/openapi/` endpoint. This commit adds
the required ProxyPass lines to the apache config file to do that.

Signed-off-by: Caleb Hill <hill@bitwise.io>

Caleb-Hill committed:
Remove empty state_delta_websockets.md file

This file was added as a placeholder and should be removed ahead of
moving over the correct file from its former location. This makes the
move a git rename not a git modification.

Signed-off-by: Caleb Hill <hill@bitwise.io>

Caleb-Hill committed:
Move rest_api docs from core to 1.2

Copy over the files from the `docs/core/1.2/rest_api/` directory to the
`docs/1.2/rest_api` directory without modification.

Signed-off-by: Caleb Hill <hill@bitwise.io>

Caleb-Hill committed:
Add contents to index.md

Construct a basic link tree linking the openapi, error_responses, and
state_delta_websockets docs together as there is not currently anywhere
to put the error_responses and state_delta_websockets files in the
sidebar.

Signed-off-by: Caleb Hill <hill@bitwise.io>

Caleb-Hill committed:
Add rest_api/index.md to sidebar

Add a link to the rest_api/index.md file in the left_sidebar file.

Signed-off-by: Caleb Hill <hill@bitwise.io>

Caleb-Hill committed:
Fix error_codes.md

This change replaces sphynx doc tables with markdown tables, and fixes
the formating on a Note. The tables look messy in raw markdown but there
really is not a way around it as markdown doesn't support multiline
table rows.

Signed-off-by: Caleb Hill <hill@bitwise.io>

Caleb-Hill committed:
Fix state_delta_websockets.md

This change fixes the formating for a Note, and multiple instances of
broken italics in the state_delta_websockets.md file.

Signed-off-by: Caleb Hill <hill@bitwise.io>

agunde406 commented on docker-compose.yaml:
```diff
@@ -34,3 +34,20 @@ services:
+    entrypoint: |
+      bash -c "
+        cd /project/docs/1.2/
+        curl https://raw.githubusercontent.com/hyperledger/sawtooth-core/1-2/rest_api/openapi.yaml -o openapi.yaml
```
We should copy the openapi.yaml file into this repo instead of pulling it in from sawtooth-core, that's what we do in grid and splinter.

Caleb-Hill replied:
Fixed

Caleb-Hill committed:
Add openapi.yaml file

This is a copy of the openapi.yaml file found in sawtooth-core, its
needed for the redoc docker container.

Signed-off-by: Caleb Hill <hill@bitwise.io>

agunde406 reviewed (dismissed):
This is missing the addition for the ci/website.dockerfile but works for `just run`. the ci update could be added in another commit/PR.

Caleb-Hill committed:
Add redoc docs to production docs container

This commit installs and runs redoc on the openapi file previously added
and then copies the output to where apache expects them to be.

Signed-off-by: Caleb Hill <hill@bitwise.io>

agunde406 commented on _includes/1.2/left_sidebar.html:
```diff
@@ -75,9 +75,9 @@
     <div class="left-sidebar-group">
         <div class="left-sidebar-header">SDK and API References</div>
-        <a href="{% link docs/1.2/sdks/python.md %}">Python</a>
+            <a href="{% link docs/1.2/sdks/python.md %}">Python</a>
```
unintended change

Caleb-Hill replied:
Fixed

agunde406 commented on docs/1.2/rest_api/index.md:
```diff
@@ -0,0 +1,13 @@
+# REST API Reference
+
```
missing 
```
<!--
  Licensed under Creative Commons Attribution 4.0 International License
  https://creativecommons.org/licenses/by/4.0/
-->
```

Caleb-Hill replied:
Fixed

Caleb-Hill committed:
Correct links to `releases` directory

The links used to point to releases/1.2/rest_api, but that was just a
placeholder for files that used to exist before the docs were moved.
This commit updates the links to point to docs/1.2/rest_api, reflows the
paragraph to keep the linter happy, and removes the placeholders.

Signed-off-by: Caleb Hill <hill@bitwise.io>
